### PR TITLE
Give ObjectEvents a useful __repr__

### PR DESCRIPTION
**Pull request by chipaca**
This change has ObjectEvents instances show useful information in
their `__repr__`, not just the default `<yadda object at 0xdeadbeef>`.

In particular, it now lists all the events the object can handle.

Fixes: #416

### CODE REVIEW

File: ops/framework.py [hunk 1]
```diff
@@ -338,23 +338,30 @@ def define_event(cls, event_kind, event_type):
         event_descriptor._set_name(cls, event_kind)
         setattr(cls, event_kind, event_descriptor)
 
-    def events(self):
-        """Return a mapping of event_kinds to bound_events for all available events."""
-        events_map = {}
+    def _event_kinds(self):
+        event_kinds = list()
```
Comment (facundobatista) [inline ops/framework.py:342]: just do `[]`

File: ops/framework.py [hunk 2]
```diff
@@ -338,23 +338,30 @@ def define_event(cls, event_kind, event_type):
...
+        return {event_kind: getattr(self, event_kind) for event_kind in self._event_kinds()}
 
     def __getitem__(self, key):
         return PrefixedEvents(self, key)
 
+    def __repr__(self):
+        k = type(self)
+        return '<{}.{}: {}>'.format(
+            k.__module__, k.__qualname__,
+            ', '.join(sorted(self._event_kinds())))
```
Comment (facundobatista) [inline ops/framework.py:363]: Shorted and easier to read?

```
        event_kinds = ', '.join(sorted(self._event_kinds()))
        return '<{}.{}: {}>'.format(k.__module__, k.__qualname__, event_kinds)
```

**Review by facundobatista — APPROVED** [review]
+1, please check two small suggestions, thanks!

**Review by balbirthomas — APPROVED** [review]
Awesome. Thank you

**Pull request merged**